fairlearn/reductions/exponentiated_gradient/_lagrangian.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import logging
+import numpy as np
+import pandas as pd
+import pickle
+import scipy.optimize as opt
+
+from fairlearn.reductions import moments
+from ._constants import _PRECISION, _INDENTATION, _LINE
+
+logger = logging.getLogger(__name__)
+
+
+class _Lagrangian:
+    """ Operations related to the Lagrangian"""
+
+    def __init__(self, dataX, dataA, dataY, learner, constraints, eps, B,
+                 opt_lambda=True, debug=False):
```

Comment:
Do you still need debug, now there's logger?

Reply:
I do not! Thank you.